### PR TITLE
build: handle login page

### LINKED ISSUE #87: handle login page

### PR DESCRIPTION
**Pull request by husamkamal**
send a request to the server 
add history to return to the last page he visits when login successfully

### CODE REVIEW

File: client/src/pages/Login/index.tsx [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
-    onSubmit: (values) => {
-      alert(JSON.stringify(values, null, 2));
+    onSubmit: async (values) => {
+      const response = await httpInstance.post('/auth/login', values);
+      if (response.data.message === 'user not found') {
+        formik.errors.email = response.data.message;
+      } else if (response.data.message === 'password not match') {
+        formik.errors.password = response.data.message;
+      } else {
+        navigate('/');
```
Comment (mai-mohammed) [inline client/src/pages/Login/index.tsx]: suggest returning the user to the previous page.

File: client/src/services/axiosCongif.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ function handleError(error: AxiosError) {
   }
   const { response }: AxiosError = error;
   const { status, data }: any = response;
+  console.log({ status, data }, '{ status, data }');
```
Comment (mai-mohammed) [inline client/src/services/axiosCongif.ts]: remove comment

**Review by mai-mohammed — CHANGES_REQUESTED** [review]

**Review by mai-mohammed — APPROVED** [review]

File: client/src/pages/Login/index.tsx [hunk 3]
```diff
@@ -1,20 +1,30 @@
 import { Button, TextField } from '@mui/material';
+import history from 'history/browser';
```
Comment (LinaYahya) [inline client/src/pages/Login/index.tsx]:
```suggestion
import history from 'history/browser'; no need as we can access history from `react-router`
```

File: client/src/pages/Login/index.tsx [hunk 4]
```diff
@@ -1,20 +1,30 @@
...
+// import { useNavigate } from 'react-router-dom';
```
Comment (LinaYahya) [inline client/src/pages/Login/index.tsx]:
```suggestion
```
Delete comments

File: server/src/controllers/login.ts [hunk 5]
```diff
@@ -2,19 +2,19 @@ import { Request } from 'express';
...
 import { generateToken } from '../helpers';
 
 const loginController = async (req:Request) => {
   const { email, password } = req.body;
   await loginSchema.validate({ email, password });
-  const result:{ password:string, id:number, fullName:string } = await loginQuery({ email });
+  const result:{ password:string, id:number, fullName:string } = await findUser({ email });
   if (!result) {
-    throw createError(400, 'user not found');
+    throw createError(400, 'there is no user ');
```
Comment (LinaYahya) [inline server/src/controllers/login.ts]:
```suggestion
    throw createError(400, 'wrong email or password ');
```

File: server/src/controllers/login.ts [hunk 6]
```diff
@@ -2,19 +2,19 @@ import { Request } from 'express';
...
   }
   const isCompare = await bcrypt.compare(password, result.password);
   if (!isCompare) {
-    throw createError(400, 'password not match');
+    throw createError(400, 'wrong password');
```
Comment (LinaYahya) [inline server/src/controllers/login.ts]:
```suggestion
    throw createError(400, 'wrong email or password ');
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by abdou059 — APPROVED** [review]

**Pull request merged**